src/stego.c

```diff
@@ -9,18 +9,26 @@
 #include <logs.h>
 #include <bmp.h>
 
+StegoMethod stego_methods[] = {
+    {"LSB1", embed_data_lsb1, retrieve_lsb1},
+    {"LSB4", embed_data_lsb4, retrieve_lsb4}
+};
```

Comment:
What if... enums...

👀

```suggestion
typedef enum {
    LSB1 = 0,
    LSB4,
    LSBI
} STEGO_ALGORITHM;

StegoMethod stego_methods[sizeof(STEGO_ALGORITHM)] = {
    {"LSB1", embed_data_lsb1, retrieve_lsb1},
    {"LSB4", embed_data_lsb4, retrieve_lsb4},
    {"LSBI", null, null}
};
```

Reply:
Solved in argp branch, no need to fix here